Assets/Scripts/Weapon.cs

```diff
@@ -0,0 +1,15 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class Weapon : MonoBehaviour
+{
+    public Enemy user;
+    public GameObject player;
```

Comment:
I don't see this used in any of the weapon derivate classes. Is it safe to remove this? It might be better to let the weapon holder (enemy/player) provide the target